### ISSUE TITLE
cy.intercept does not properly stub when there are request headers

### ISSUE THREAD

**Issue by tgugli** [cypress-io/cypress#9602]
### Current behavior
I am trying to stub out some API calls and I do not want to actually reach the API server. I am seeing that whenever I make a cross domain request with request headers the request actually reaches the destination server. In my example since I am trying to fetch from a non-existent domain the request times out.

### Desired behavior
The request should be stubbed without hitting the destination server. 

### Test code to reproduce
```js
it('Stubbing cross domain without headers works', () => {
  cy.intercept('get', '/endpoint', { body: 'hello' }).as('getUrl')
  cy.visit('https://example.com')
  cy.window().then(win => {
    fetch('https://someotherapiserverthatdoesntreallyexist.com/endpoint', {
      method: 'get'
    })
  })
  cy.wait('@getUrl')
})

it('Stubbing cross domain with headers does not work', () => {
  cy.intercept('get', '/endpoint', { body: 'hello' }).as('getUrl')
  cy.visit('https://example.com')
  cy.window().then(win => {
    fetch('https://someotherapiserverthatdoesntreallyexist.com/endpoint', {
      method: 'get',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer token'
      }
    })
  })
  cy.wait('@getUrl')
})
```

### Versions

Version 6.1.0

**Comment by bahmutov** [comment]
Added reproducible example in https://github.com/cypress-io/cypress-example-recipes/pull/606

**Comment by jennifer-shehane** [comment]
This has been resolved and is no longer present in 6.2.1.

**Comment by jennifer-shehane** [comment]
Revisiting this. This was fixed in 6.2.0.
